test/deadline_submitter_for_houdini/unit/test_assets.py

```diff
@@ -212,3 +219,248 @@ def test_parse_files_manually_added(
+
+    # THEN
+    assert actual == expected
+
+
+# "pattern", "filenames"
+filenames_with_frames = [
+    param(
+        "sequence.$F.png",
+        [
+            "sequence.0.png",
+            "sequence.1.png",
+        ],
+        id="$F",
+    ),
+    param(
+        "sequence.${F}.png",
+        [
+            "sequence.0.png",
+            "sequence.1.png",
+        ],
+        id="Protected $F",
+    ),
+    param(
+        "sequence.$F4.png",
+        ["sequence.0000.png", "sequence.0001.png", "sequence.0002.png"],
+        id="$F with padding",
+    ),
+    param(
+        "sequence.${F4}.png",
+        ["sequence.0000.png", "sequence.0001.png", "sequence.0002.png"],
+        id="Protected $F with padding",
+    ),
+    param(
+        "sequence.$F4.$F5.png",
+        [
+            "sequence.000.0000.png",
+            "sequence.0001.00001.png",
+        ],
+        id="Multiple $F with padding",
+    ),
+    param(
+        "sequence.${F}F.png",
+        [
+            "sequence.0F.png",
+            "sequence.1F.png",
+        ],
+        id="Protected $F with trailing F",
+    ),
+    param(
+        "sequence.${F4}F.png",
+        [
+            "sequence.0000F.png",
+            "sequence.0001F.png",
+        ],
+        id="Protected $F with padding and trailing F",
+    ),
+    param(
+        "sequence.$FF.png",
+        [
+            "sequence.0.png",
+            "sequence.01.png",
+        ],
+        id="$FF",
+    ),
+    param(
+        "sequence.${FF}.png",
+        [
+            "sequence.0.png",
+            "sequence.01.png",
+        ],
+        id="Protected $FF",
+    ),
+    param(
+        "sequence.$FFF.png",
+        [
+            "sequence.0F.png",
+            "sequence.1F.png",
+        ],
+        id="$FF with trailing F",
+    ),
+    param(
+        "sequence.$SF.png",
+        [
+            "sequence.0.png",
+            "sequence.1.png",
+        ],
+        id="$SF",
+    ),
+    param(
+        "sequence.${SF}.png",
+        [
+            "sequence.0.png",
+            "sequence.1.png",
+        ],
+        id="Protected $SF",
+    ),
+    param(
+        "sequence.$ST.png",
+        [
+            "sequence.0.png",
+            "sequence.1.png",
+        ],
+        id="$ST",
+    ),
+    param(
+        "sequence.${ST}.png",
+        [
+            "sequence.0.png",
+            "sequence.1.png",
+        ],
+        id="Protected $ST",
+    ),
+    param(
+        "sequence.`python stuff`.png",
+        [
+            "sequence.0.png",
+            "sequence.1.png",
+        ],
+        id="Python expression",
+    ),
+    param(
+        "sequence.`$F5`.png",
+        [
+            "sequence.0.png",
+            "sequence.1.png",
+        ],
+        id="Python expression with a frame variable within",
+    ),
+]
+
+
+@pytest.mark.parametrize(("pattern", "filenames"), filenames_with_frames)
+def test_filenames_with_frames(tmpdir, pattern: str, filenames: list[str]):
```

Comment:
Might be missing it, but it looks like none of these test cases include additional files that **shouldn't** be included in the matching. Can an additional parameter be added for `expected_files `and additional files added to the `filenames` that shouldn't/won't be matched?